credentialdigger/cli/hook.py

```diff
@@ -0,0 +1,180 @@
+  --rules RULES      Specify the yaml file path containing the scan rules
+                     e.g., /path/to/rules.yaml
+  --db_path DB_PATH  Specify the database file path where to save the results
+                     e.g., /path/to/data.db
+  --no_interaction   Flag used to remove the interaction i.e.,
+                     do not prompt if the commit should continue
+                     in case of discoveries. If specified, the hook will
+                     fail in case of discoveries.
+"""
+
+import subprocess
+import sys
+from pathlib import Path
+
+from credentialdigger import SqliteClient
+from credentialdigger.models.model_manager import ModelManager
+
+
+def configure_parser(parser):
+    """ Configure arguments for command line parser.
+
+    Parameters
+    ----------
+    parser: `credentialdigger.cli.customParser`
+        Command line parser
+    """
+    parser.set_defaults(func=run)
+
+
+def system(*args, **kwargs):
+    """Run a command and get the result."""
+    kwargs.setdefault('stdout', subprocess.PIPE)
+    proc = subprocess.Popen(args, **kwargs)
+    out, err = proc.communicate()
+    return out
+
+
+def print_msg(msg):
+    """Print messages to /dev/tty"""
+    subprocess.run(f'echo \"\n{msg}\n\" > /dev/tty',
+                   shell=True,
+                   stdout=subprocess.PIPE)
+
+
+def ask_commit(str_discoveries):
+    """Ask for the commit confirmation in case of possible leaks"""
+
+    msg = 'You have the following disoveries:\n' \
+          f'{str_discoveries}\ncontinue? (y/n)'
+    print_msg(msg)
+
+    sys.stdin = open('/dev/tty', 'r')
+    # Create a process on /dev/tty to capture the input (commit or not)
+    # It reads the input, saves it in userinput and echos it
+    # subprocess.check_output return the output of the command i.e., userinpput
+    user_input = subprocess.check_output('read -p \"\" userinput && echo '
+                                         '\"$userinput\"',
+                                         shell=True, stdin=sys.stdin).rstrip()
+
+    return user_input.decode('utf-8')
+
+
+def run(args):
+    """Run credential digger on staged files
+
+    Parameters
+    ----------
+    args: `argparse.Namespace`
+        Arguments from command line parser.
+
+    Returns
+    -------
+        It exits with success code 0 if there are no discoveries, otherwise
+        it exits with an error code != 0 printing the leaks
+    """
+
+    files_status = system('git', 'diff', '--name-status', '--staged'
+                          ).decode('utf-8').splitlines()
+    files = []
+    for fs in files_status:
+        stats = fs.split()
+        status = stats[0]
+        # Takes the last filename which is, in case of renamed files,
+        # the new name
+        filename = stats[-1]
+        # D = deleted files
+        # R = renamed files
+        # It considers the first char because the renamed files are displayed
+        # as Rxxx where xxx is a number
```

Comment:
Does `status` have a limited number of possible values? is it only for renamed files that the format is Rxxx or also for others (e.g., Dxxxx)